docs/manual/developer_guide.adoc

```diff
@@ -1334,13 +1334,109 @@ Available low-level Jinja macros that can be used in Bash remediations:
 - `die` - Function to terminate the remediation
 - `set_config_file` - Add an entry to a text configuration file
 
-==== Templating
+
+==== Using Templates
+
+Writing OVAL checks, Bash, or any other content can be tedious work. For
+certain types of rules we provide templates. If there is a template that can be
+used for the new rule you only specify the template name and its parameters in
+`rule.yml` and the content will be generated during the build.
+
+The templating system currently supports OVAL checks and Ansible, Bash,
+Anaconda and Puppet remediations.  All templates can be found in
+`shared/templates` directory. The files are named `template_<TYPE>_<NAME>`,
+where `<TYPE>` should be OVAL, ANSIBLE, BASH, ANACONDA or PUPPET and `<NAME>`
+is the template name.
+
+To use a template in `rule.yml` add `template:` key there and fill it
+accordingly. The general form is the following:
+
+----
+template:
+    name: template_name
+    vars:
+        param_name: value # these parameters individual for each template
+        param_name@rhel7: value1
+        param_name@rhel8: value2
+    backends: # optional
+        ansible: off
+        bash: on # on is implicit value
+----
+
+The `vars:` key contain template parameters and their values which will be
+substituted into the template.  Each template has specific parameters.  To use
+different values of parameters based on product, append `@` followed by product
+ID to the parameter name.
+
+The `backends` key is optional. By default, every language for which a template
+with given name exist will be generated. This key can be used to explicitly opt
+out from generating a certain type of content for the rule.
+
+For example, to generate templated content except Bash remediation for rule
+"Package GCC is Installed" using `package_installed` template, add the
+following to `rule.yml`:
+
+----
+template:
+    name: package_installed
+    vars:
+        pkgname: gcc
+    backends:
+        bash: off
```

Comment:
Unfortunately we have to quote the boolean.

Reply:
That's strange, I have tested using `off` without quoting and the build system was able to switch off correctly the templated content. Do you remember what was the scenario where quoting is required?

Reply:
@ggbecker The problem is when you use Python 2 and you don't use quotes you will get a traceback due to some explicit "unicode" typecast in the generated processed rule.yml YAML file. 

For this moment, we will use quotes in the documentation.

But in future I wish to remove https://github.com/ComplianceAsCode/content/blob/master/ssg/yaml.py#L27 , which is causing this, but it needs to be investigated what else it could possibly break.